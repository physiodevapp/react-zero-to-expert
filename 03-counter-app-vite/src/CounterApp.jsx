import PropTypes from 'prop-types'

function CounterApp({value}) {
  // console.log(value)
  return (
    <>
      <h1>CounterApps</h1>
      <h2>{value}</h2>
    </>
  )
}

CounterApp.propTypes = {
  value: PropTypes.number.isRequired 
}

CounterApp.defaultProps = {
  value: 0
}

export default CounterApp