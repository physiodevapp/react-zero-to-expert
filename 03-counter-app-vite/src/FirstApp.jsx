import PropTypes from "prop-types";

export const FirstApp = ({ title, subtitle }) => {
  return (
    <>
      <h1 data-testid='test-title'>{title}</h1>
      <p>{subtitle}</p>
      <p>{subtitle}</p>
    </>
  );
};

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.number.isRequired,
};

FirstApp.defaultProps = {
  // title: "ups!",
  subtitle: 123,
  name: "Physiodevapp",
};
