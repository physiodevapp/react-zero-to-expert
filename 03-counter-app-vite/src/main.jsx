import React from 'react';
import ReactDOM from 'react-dom/client';
import { HelloWorldApp } from './HelloWorldApp';
// import App from './HelloWorldApp'

ReactDOM.createRoot( document.getElementById('root') ).render(
  <React.StrictMode>
    <HelloWorldApp/>
  </React.StrictMode>
)