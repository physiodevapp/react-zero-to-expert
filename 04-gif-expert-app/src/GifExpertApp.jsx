import React, { useState } from "react";
import AddCategory from "./components/AddCategory";

function GifExpertApp() {
  const [categories, setCategories] = useState(["One Punch", "Dragon Ball"]);

  const handleAddCategory = () => {
    setCategories((prevCategories) => {
      return ['Valorant', ...prevCategories]
    })
  };

  return (
    <>
      {/* title */}
      <h1>GifExpertApp</h1>

      {/* input */}
      <AddCategory/>

      {/* item-list */}

      <button onClick={handleAddCategory}>Add category</button>

      <ol>
        {categories.map((category) => (
          <li key={category}>{category}</li>
        ))}
      </ol>

      {/* item */}
    </>
  );
}

export default GifExpertApp;
