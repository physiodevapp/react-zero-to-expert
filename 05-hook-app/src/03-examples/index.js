export * from './MultipleCustomHooks'
export * from './LoadingQuote'
export * from './Quote'