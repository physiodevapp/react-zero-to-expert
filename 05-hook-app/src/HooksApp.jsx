import React from 'react'

function HooksApp() {
  return (
    <h2>HooksApp</h2>
  )
}

export default HooksApp