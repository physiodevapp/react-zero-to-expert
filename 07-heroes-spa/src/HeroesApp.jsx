import React from "react";
import { AppRouter } from "./router/AppRouter";
import { AuthProvider } from "./heroes/context";

export const HeroesApp = () => {
  return (
    <AuthProvider>
      <AppRouter />
    </AuthProvider>
  );
};
