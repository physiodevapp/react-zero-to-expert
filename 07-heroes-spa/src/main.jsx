import React from "react";
import ReactDOM from "react-dom/client";
import "./styles.css";
import { HeroesApp } from "./HeroesApp";
import { BrowserRouter as Router } from "react-router-dom";

ReactDOM.createRoot(document.getElementById("root")).render(
  <React.StrictMode>
    <Router>
      <HeroesApp />
    </Router>
  </React.StrictMode>
);
