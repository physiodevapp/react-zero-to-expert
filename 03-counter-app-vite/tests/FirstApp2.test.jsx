import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe("Testing <FirstApp/>", () => {
  const title = "Hola soy Goku";
  const subtitle = 123;

  test("should match snapshot", () => {
    const { container } = render(<FirstApp title={title} />);

    expect(container).toMatchSnapshot();
  });

  test('should show "Hola soy Goku"', () => {
    render(<FirstApp title={title} />);
    // screen.debug()
    expect(screen.getByText(title)).toBeTruthy();
  });

  test("should show title in h1", () => {
    render(<FirstApp title={title} />);
    expect(screen.getByRole("heading", { level: 1 }).innerHTML).toBe(title);
  });

  test("should show subtitle by props", () => {

    render(<FirstApp title={title} subtitle={subtitle}/>);

    expect(screen.getAllByText(subtitle).length).toBe(2);

  });
});
