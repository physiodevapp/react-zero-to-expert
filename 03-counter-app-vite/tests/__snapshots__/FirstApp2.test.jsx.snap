// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <FirstApp/> should match snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola soy Goku
  </h1>
  <p>
    123
  </p>
  <p>
    123
  </p>
</div>
`;
